core/Pos/Binary/Core/Types.hs

```diff
@@ -61,3 +61,32 @@ instance Bi T.SharedSeed where
 instance Bi T.ChainDifficulty where
     get = label "ChainDifficulty" $ T.ChainDifficulty . getUnsignedVarInt <$> get
     put (T.ChainDifficulty c) = put (UnsignedVarInt c)
+
+instance Bi T.BlockVersionData where
+    get = label "BlockVersionData" $ do
```

Comment:
> if it doesn't decrease performance

It does, afaik. At least for Aeson, TH-derived instances are 2× faster or something.